src/ae.c

```diff
@@ -484,14 +484,24 @@ int aeWait(int fd, int mask, long long milliseconds) {
     if (mask & AE_READABLE) pfd.events |= POLLIN;
     if (mask & AE_WRITABLE) pfd.events |= POLLOUT;
 
-    if ((retval = poll(&pfd, 1, milliseconds))== 1) {
-        if (pfd.revents & POLLIN) retmask |= AE_READABLE;
-        if (pfd.revents & POLLOUT) retmask |= AE_WRITABLE;
-        if (pfd.revents & POLLERR) retmask |= AE_WRITABLE;
-        if (pfd.revents & POLLHUP) retmask |= AE_WRITABLE;
-        return retmask;
-    } else {
-        return retval;
+    monotime begin = getMonotonicUs();
```

Comment:
getMonotonicUs() here involves an unexpected dependency on monotonicInit(), but the caller may be not aware of it, now the CI failes  because redis-cli doesn't initialize monotonic clock. There're three possible ways:
1. monotonicInit in redis-cli.c
2. monotonicInit here, i.e. aeWait
3. let the caller of aeWait  address the case of EINTR, i.e. retry safely without a precise timeout.

Reply:
@uvletter I think it is fine to request that user will issue call to monotonicInit before using the clock API (me saying I am fine with option 1) or it can also be lazy initialized (ie option 2) 
in any case I think it would be great to keep a static flag indicating init was done in order to lazy initialize(option 2) or asset with much more clear reason - what do you think?

Reply:
I'm ok with both option 1 and 2, personally I like 2 more since a library function'd better not expose the dependency to caller. If we finally decide to choose option 1, I agree we need to make the dependency more explicit, by an assert with much more clear reason, or a comment on top of function or something else.

> in any case I think it would be great to keep a static flag indicating init was done in order to lazy initialize(option 2)

Actually monotonicInit is safe to be involked many times as it checks if initialization is done, an inelegant but more effecient way is to check if getMonotonicUs is NULL and then call monotonicInit, which can save a function call at most time.